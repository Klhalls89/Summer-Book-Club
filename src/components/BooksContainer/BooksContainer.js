import React from 'react';
import BookCard from '../BookCard/BookCard'


const BooksContainer = (props) => {
  const showBooks = props.books.map(book => <BookCard {...book} key={book.id} />
  )
  return(
    <div>
      BooksContainer
      {showBooks}
    </div>
  );
};

export default BooksContainer;
