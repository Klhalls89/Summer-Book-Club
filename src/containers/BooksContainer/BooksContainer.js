import React, { Component } from 'react';
import BookCard from '../../components/BookCard/BookCard'
import { connect } from 'react-redux'



class BooksContainer extends Component {
  constructor() {
    super()
  }

  render() {
    const showBooks = this.props.books.map(book => <BookCard {...book} key={book.id} />
  )
    return(
    <div className='books-container'>
      {showBooks}
    </div>
    );
  }  
};

export const mapStateToProps = (state) => ({
  books: state.books
})

export default connect(mapStateToProps)(BooksContainer);
