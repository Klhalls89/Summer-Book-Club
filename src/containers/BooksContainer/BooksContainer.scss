.books-container {
  margin-top: 10px;
  display: grid;
  grid-template-columns: repeat(6, 1fr);
  grid-gap: 10px;
}