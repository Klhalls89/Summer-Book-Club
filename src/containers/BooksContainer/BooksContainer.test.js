import React from "react";
import { BooksContainer } from './BooksContainer';
import { mapStateToProps } from './BooksContainer';
import { shallow } from 'enzyme';


describe('BooksContainer', () => {
  let wrapper;
  let mockBooks;

  beforeEach(() => {
    mockBooks = [{ title: "the best one"}];
    wrapper = shallow(<BooksContainer books={mockBooks}/>);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {

    it('should return a props object from state', () => {
      const mockState = {
        loading: false,
        books: [],
        error: 'error'
      }
      const expected = {
        books: []
      }
      const results = mapStateToProps(mockState)
      expect(results).toEqual(expected)

    });
  });
});