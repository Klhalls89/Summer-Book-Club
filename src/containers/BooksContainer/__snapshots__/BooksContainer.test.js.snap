// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksContainer should match snapshot 1`] = `ShallowWrapper {}`;
